internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/utils/ConfigParser.java

```diff
@@ -0,0 +1,248 @@
+import org.wso2.security.tools.scanner.dependency.js.reportpublisher.ReportUploader;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+import java.util.Random;
+
+/**
+ * Util class to parse configuration properties. There are multiple configuration files related to each product and
+ * credentials related configuration files. Product related configuration files are parsed during preprocessing stage
+ * and credentials related config files are passed at the time where it is to be used.
+ */
+public class ConfigParser {
+
+    private static final Logger log = Logger.getLogger(ConfigParser.class);
+    private static Random random = new Random();
+
+    private ConfigParser() {
+    }
+
+    /**
+     * Parse the list of products which supports for weekly scanning.
+     *
+     * @return list of product names.
+     * @throws ConfigParserException exception occurred while parsing configuration details.
+     */
+    public static List<String> parseProductList() throws ConfigParserException {
+        List<String> products;
+        Properties properties = new Properties();
+        ClassLoader classLoader = ConfigParser.class.getClassLoader();
+        //load a properties file from class path
+        try (InputStream input = classLoader.getResourceAsStream(JSScannerConstants.PRODUCT_LIST_FILE)) {
+            properties.load(input);
+        } catch (IOException e) {
+            throw new ConfigParserException("Error occurred in parsing productList config file : " + e);
+        }
+        products = Arrays.asList(properties.getProperty(JSScannerConstants.PRODUCTS).
+                split(JSScannerConstants.PROPERTIES_FILE_DELIMETER));
+        log.info("[JS_SEC_DAILY_SCAN] Supported products are : " + String.join(",", products));
+        return products;
+    }
+
+
+    /**
+     * This method is responsible to parse properties regarding to upload scan reports into particular github
+     * repository. THe following properties are read from config file.
+     * 1.Username
+     * 2.Password
+     * 3.Repository url where the reports should be uploaded.
+     *
+     * @throws ConfigParserException exception occurred while parsing configuration details.
+     */
+    public static ReportUploader parseGitUploaderConfigProperties() throws GitAPIException, ConfigParserException {
+        Properties properties = new Properties();
+        ClassLoader classLoader = ConfigParser.class.getClassLoader();
+        try (InputStream input = classLoader.getResourceAsStream(JSScannerConstants.GIT_CONFIG_FILE)) {
+            properties.load(input);
+        } catch (IOException e) {
+            throw new ConfigParserException("Error occurred in parsing github credentials : " + e);
+        }
+        ReportUploader reportUploader = new GitUploader(properties.getProperty(JSScannerConstants.USERNAME)
```

Comment:
parseGitUploaderConfigProperties method name gives the idea that this only reads properties related to git. However, constructing 'reportUploader' is much more than just reading properties. It involves calling git APIs as well. Please redesign this. Methods need to be cohesive, and method name should clearly give the idea of the provided functionality.